app/controllers/SavedProgressController.scala

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import com.typesafe.play.cachecontrol.Seconds
+import config.FrontendAppConfig
+import controllers.actions._
+
+import javax.inject.Inject
+import models.Period
+import play.api.i18n.I18nSupport
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendBaseController
+import views.html.SavedProgressView
+
+import java.time.{LocalDate, ZoneId}
+import java.time.format.DateTimeFormatter
+import java.time.temporal.ChronoUnit
+
+class SavedProgressController @Inject()(
+                                       cc: AuthenticatedControllerComponents,
+                                       view: SavedProgressView,
+                                       appConfig: FrontendAppConfig
+                                     ) extends FrontendBaseController with I18nSupport {
+
+  protected val controllerComponents: MessagesControllerComponents = cc
+
+  def onPageLoad(period: Period): Action[AnyContent] =
+    (cc.actionBuilder andThen cc.identify andThen cc.getRegistration andThen cc.getData(period) andThen cc.requireData) {
+    implicit request =>
+      val dayFormatter  = DateTimeFormatter.ofPattern("d MMMM yyyy")
```

Comment:
I'm not sure if there is already a formatter of this in formats/Format.scala if not, could you add it there and use in test as well?

Reply:
yeah there was a formatter, thank you! done now